### PR TITLE
ROX-20718: ComplianceCheckResults Add Warnings, ValuesUsed, Rationale

### PR DESCRIPTION
**Pull request by Vincent056**
## Description

Add rationale, warnings, and values used to the storage, api, and internalapi protos.
Populate the values in sensor

## Checklist
- [ ] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Unit testing

### Here I tell how I validated my change

* Added test in [complianceoperatorresults/pipeline_test.go](https://github.com/stackrox/stackrox/pull/9070/files#diff-4b0627931e85178f6ddfca484839c8e7d5ee11ae086a3b94c0bfe46fadd096f2) to test `ValuesUsed` and `Warnings` conversion

### CODE REVIEW

File: proto/internalapi/central/compliance_operator.proto [hunk 1]
```diff
@@ -152,6 +152,9 @@ message ComplianceOperatorCheckResultV2 {
     string                         control      = 13;
     string                         scan_name    = 14;
     string                         suite_name   = 15;
+    string                         rationale    = 16;
```
Comment (dashrews78) [inline proto/internalapi/central/compliance_operator.proto:155]: Update the `// Next tag:` comment at the top of the message

File: proto/api/v2/compliance_results_service.proto [hunk 2]
```diff
@@ -41,6 +41,9 @@ message ComplianceCheckResult {
     string                      instructions = 5;
     string                      standard     = 6;
     string                      control      = 7;
+    string                      rationale    = 8;
```
Comment (dashrews78) [inline proto/api/v2/compliance_results_service.proto:44]: Add a `// Next tag:` comment at the top of this message

File: proto/storage/compliance_operator_v2.proto [hunk 3]
```diff
@@ -73,6 +73,8 @@ message ComplianceOperatorRuleV2 {
     string              description      = 9;
     string              rationale        = 10;
     string              fixes            = 11;
+    repeated string     valuesUsed       = 12;
+    repeated string     warnings         = 13;
```
Comment (dashrews78) [inline proto/storage/compliance_operator_v2.proto:77]: Update the `// Next tag:` comment at the top of the message

File: proto/storage/compliance_operator_v2.proto [hunk 4]
```diff
@@ -139,6 +141,9 @@ message ComplianceOperatorCheckResultV2 {
     string                    control          = 13;
     string                    scan_id          = 14 [(gogoproto.moretags) = 'search:"-" sql:"fk(ComplianceOperatorScanV2:id),no-fk-constraint"'];
     string                    scan_config_name = 16 [(gogoproto.moretags) = 'search:"Compliance Scan Name,store" sql:"fk(ComplianceOperatorScanConfigurationV2:scan_name),no-fk-constraint"'];
+    string                    rationale        = 17;
```
Comment (dashrews78) [inline proto/storage/compliance_operator_v2.proto:144]: Update the `// Next tag:` comment at the top of the message

**Review by dashrews78 — CHANGES_REQUESTED** [review]
very minor but important comments in the protos.

File: proto/api/v2/compliance_results_service.proto (hunk 2 above)
Comment (Vincent056) [reply proto/api/v2/compliance_results_service.proto:44]: that's a good catch!

**Review by dashrews78 — APPROVED** [review]

**Pull request merged**